### DISCUSSION TITLE
twin size an twin area

### DISCUSSION THREAD

**Discussion by lwlgeo** [mtex-toolbox/mtex#1817] [Ask Anything]
hi, everyone,
i am trying to calcualted twin size and twin area in a deformed materials. however, the example in the mtex is showing the sizes of twined grains, no the twin itself, the code is showed as :
twinId = unique(gB_MgMg(isTwinning).grainId);
% compute the area fraction
sum(area(grains(twinId))) / sum(area(grains)) * 100

through this,  i can get the relative area of these initial grains that have undergone twinning to total area. however, what i really want to know is the fraction of the twins in the whole map, not the grains undergone twinning.

is anyone know hot to extract the information of twin grains?

Thanks a lot

**Comment by marcoalopez** [comment]
Hi,

 to calculate how much of your grain boundaries are twins I think the most direct way is to calculate the ratio between the length of Mg-Mg grain boundaries that are twins and those that are not. For example:

```matlab
% calc twin interface fraction
100 * sum(Mg_twin_GB.segLength) / sum(Mg_Mg_GB.segLength)
```

where ``Mg_twin_GB`` would represent the Mg-Mg grain boundaries that are twins and ``Mg_Mg_GB`` all the Mg-Mg grain boundaries.  The command ``segLength`` returns the length of boundary segments in µm, see https://mtex-toolbox.github.io/grainBoundary.segLength.html

Regarding the twinned area, unless you know the orientation of the grains prior to the twinning, I don't think it is safe to calculate it because how do you know which part of the grain belongs to the original (untwinned) orientation and to the twinned one? For some, it may be obvious, but for many, you will not be able to tell I think.

Cheers.

**Reply by lwlgeo** [reply]
Hi, Lopez-Sanchez,
thank you very much for your kindly reply.
here is my thought. we dont need to know the orientation of the grains prior to the twinning, because the grain boundaries of the twin is dominated by the twin boundary, while it not the case for the rest. what we need is setting a threshold for the ratio of twin boundary to define the twin, while the rest is the area with no twin?

Best

**Comment by marcoalopez** [comment]
Hi again. You're welcome.

By not knowing which part of the grain is twinned, I mean in cases where a grain is crossed by a twin plane, in which case you can't know which part was twinned unless you know the original orientation of the grain. This is why I usually use the % of grain boundaries that are twinned to those that are not, rather than the twinned area. But if in your case all the twinned areas are bounded by twins, you won't have a problem.

Best.
Marco
